.github/workflows/release.yml

```diff
@@ -0,0 +1,22 @@
+name: Create Release
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  build:
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Hatch
+        run: pipx install hatch
+      - name: Run tests
+        run: hatch run test:test
+      - name: Build dist
+        run: hatch build
+      - name: Publish on PyPI
+        if: github.ref == 'refs/heads/main'
+        run: hatch publish
```

Comment:
maybe we can pull some inspiration by how this works:
https://github.com/pypa/hatch/blob/8e0db522be51f47e189b9357c5d11b03787ca6a7/.github/workflows/build-hatch.yml#L440-L483

Some things that I like here that we should take inspiration from:
* breaks down the steps into a few different jobs (separation of build, test, publish). This makes for better retries / observability into a failed workflow.
* OIDC auth for publishing to PyPI. More secure than using secrets. https://github.com/marketplace/actions/pypi-publish#trusted-publishing
* GitHub action is responsible for creating the release itself https://github.com/softprops/action-gh-release (so our wheel artifact is actually part of the GitHub release)

Reply:
Updated; I've broken the pipeline into build vs publish, but keeping the release as manual so we don't have to automate the release notes.